### PR TITLE
qa: check for rank hole without racy joinable flag gymnastics

### PR DESCRIPTION
**Pull request by batrick**
Fixes: https://tracker.ceph.com/issues/48770
Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by batrick** [comment]
jenkins test make check

File: qa/tasks/cephfs/test_failover.py [hunk 1]
```diff
@@ -265,26 +265,23 @@ def test_hole(self):
...
 
+        # Now add a delay which should slow down how quickly rank 1 stops
+        self.config_set('mds', 'ms_inject_delay_max', '5.0')
+        self.config_set('mds', 'ms_inject_delay_probability', '1.0')
         self.fs.set_max_mds(1)
         log.info("status = {0}".format(self.fs.status()))
 
-        self.fs.set_max_mds(3)
         # Don't wait for rank 1 to stop
+        self.fs.set_max_mds(3)
```
Comment (vshankar) [inline qa/tasks/cephfs/test_failover.py:275]: nit: could `grow/shrink` filesystem helpers be used here with some tweaks to not wait for daemons to be healthy?

Comment (batrick) [reply qa/tasks/cephfs/test_failover.py:275]: I kinda like how the `set_max_mds` is explicit about what it's doing. `grow`/`shrink` may do other things in the future which could disrupt this test.

Comment (vshankar) [reply qa/tasks/cephfs/test_failover.py:275]: fair enough.

**Comment by batrick** [comment]
https://pulpito.ceph.com/pdonnell-2021-01-08_05:07:01-fs-wip-pdonnell-testing-20210107.220115-distro-basic-smithi/

New failures are:
- https://tracker.ceph.com/issues/48811

**Comment by batrick** [comment]
Ready for merge.

**Review by vshankar — APPROVED** [review]

**Pull request merged**